test_conformance/basic/test_vector_swizzle.cpp

```diff
@@ -0,0 +1,677 @@
+
+    std::string buildOptions{ "-DTYPE=" };
+    buildOptions += type_name;
+    buildOptions += std::to_string(N);
+    buildOptions += " -DBASETYPE=";
+    buildOptions += type_name;
+
+    std::vector<T> value(N);
+    std::iota(value.begin(), value.end(), 0);
+
+    std::vector<T> reference;
+    constexpr size_t S = TestInfo<N>::vector_size;
+    makeReference<T, N, S>(reference);
+
+    // XYZW swizzles:
+
+    const char* xyzw_source = TestInfo<N>::kernel_source_xyzw;
+    error = create_single_kernel_helper(
+        context, &program, &kernel, 1, &xyzw_source, "test_vector_swizzle_xyzw",
+        buildOptions.c_str());
+    test_error(error, "Unable to create xyzw test kernel");
+
+    result |= test_vectype_case(value, reference, context, kernel, queue);
+
+    // sN swizzles:
+    const char* sN_source = TestInfo<N>::kernel_source_sN;
+    error = create_single_kernel_helper(context, &program, &kernel, 1,
+                                        &sN_source, "test_vector_swizzle_sN",
+                                        buildOptions.c_str());
+    test_error(error, "Unable to create sN test kernel");
+
+    result |= test_vectype_case(value, reference, context, kernel, queue);
+
+    // RGBA swizzles for OpenCL 3.0 and newer:
+    const Version device_version = get_device_cl_version(device);
+    if (device_version >= Version(3, 0))
+    {
+        const char* rgba_source = TestInfo<N>::kernel_source_rgba;
+        error = create_single_kernel_helper(
+            context, &program, &kernel, 1, &rgba_source,
+            "test_vector_swizzle_rgba", buildOptions.c_str());
+        test_error(error, "Unable to create rgba test kernel");
+
+        result |= test_vectype_case(value, reference, context, kernel, queue);
+    }
+
+    return result;
+}
+
+template <typename T>
+static int test_type(const char* type_name, cl_device_id device,
+                     cl_context context, cl_command_queue queue)
+{
+    return test_vectype<T, 2>(type_name, device, context, queue)
+        | test_vectype<T, 3>(type_name, device, context, queue)
+        | test_vectype<T, 4>(type_name, device, context, queue)
+        | test_vectype<T, 8>(type_name, device, context, queue)
+        | test_vectype<T, 16>(type_name, device, context, queue);
+}
+
+int test_vector_swizzle(cl_device_id device, cl_context context,
+                        cl_command_queue queue, int num_elements)
+{
+    int hasDouble = is_extension_available(device, "cl_khr_fp64");
+
+    int result = TEST_PASS;
+    result |= test_type<cl_char>("char", device, context, queue);
+    result |= test_type<cl_uchar>("uchar", device, context, queue);
+    result |= test_type<cl_short>("short", device, context, queue);
+    result |= test_type<cl_ushort>("ushort", device, context, queue);
+    result |= test_type<cl_int>("int", device, context, queue);
+    result |= test_type<cl_uint>("uint", device, context, queue);
+    result |= test_type<cl_long>("long", device, context, queue);
```

Comment:
Please add conditionally this data type - gHasLong from testHarness

Reply:
Added - thanks!

Reply:
but `cl_float` should not be there ...

Reply:
Oh dear, that's an unfortunate copy-paste error.  Fixed!